test/extended/ha/ha.go

```diff
@@ -0,0 +1,258 @@
+package ha
+
+/*
+This controllers test suite is not part of the "default" group, because its
+testing involves terminating and launching controllers components of OpenShift.
+Therefore it can't be run in parallel.
+*/
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("ha: Election of OpenShift controllers", func() {
+	defer g.GinkgoRecover()
+	var (
+		podConfig  = exutil.FixturePath("fixtures", "hello-world-pod.json")
+		reExitCode = regexp.MustCompile(`(?i)exitcode:(\d+)`)
+	)
+
+	// Run a controller manager with one intitial controllers instance.
+	// Running controllers are needed for NewCLI to run.
+	// FIXME: seems like OutputDir is unset at this moment (execution of the first test)
+	m, _err := exutil.NewControllersManager(exutil.MasterConfigPath(), 0, 8444, exutil.TestContext.OutputDir)
+	if _err != nil {
+		g.Fail(fmt.Sprintf("Failed to initialize controllers instance: %v\n", _err))
+	}
+	// FIXME: Controllers instance immediatelly terminates after acquiring its lease without following sleep
+	time.Sleep(1 * time.Second)
+
+	// One controllers instance needs to be running before CLI can be initialized
+	_, _err = m.StartNewInstance()
+	if _err != nil {
+		g.Fail(fmt.Sprintf("Failed to initialize controllers instance: %v\n", _err))
```

Comment:
It seems that Ω can't be used outside `(Before|After)Each|It`:

    Test Panicked
    You are trying to make an assertion, but Gomega's fail handler is nil.
    If you're using Ginkgo then you probably forgot to put your assertion in an It().
    Alternatively, you may have forgotten to register a fail handler with RegisterFailHandler() or RegisterTestingT().